application/classes/Ushahidi/Listener/IntercomAdminListener.php

```diff
@@ -0,0 +1,55 @@
+<?php defined('SYSPATH') or die('No direct script access');
+
+/**
+ * Ushahidi Intercom Listener
+ *
+ * Listens for new posts that are added to a set
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+use League\Event\AbstractListener;
+use League\Event\EventInterface;
+
+use Intercom\IntercomClient;
+
+use GuzzleHttp\Exception\ClientException;
+
+class Ushahidi_Listener_IntercomAdminListener extends AbstractListener
+{
+
+  public function handle(EventInterface $event, $user = null)
+  {
+    if($user && $user->role === 'admin') {
+      $intercomAppToken = getenv('INTERCOM_APP_TOKEN');
+      $domain = service('site');
```

Comment:
This should return `quakemap.api.ushahidi.io` or just an empty string if we're *not* on an ushahidi.io deployment ... was that what you expected?